src/stream/src/common/log_store_impl/kv_log_store/reader.rs

```diff
@@ -375,6 +376,44 @@ impl<S: StateStoreRead> KvLogStoreReader<S> {
 }
 
 impl<S: StateStoreRead> LogReader for KvLogStoreReader<S> {
+    async fn start_from(&mut self, start_offset: Option<u64>) -> LogStoreResult<()> {
+        // init or rewind must be executed before start_from.
+        let aligned_range_start =
+            if let KvLogStoreReaderFutureState::Reset(aligned_range_start) = &self.future_state {
+                aligned_range_start
+            } else {
+                panic!("future state is not Reset");
+            };
+        // still consuming persisted state store data
+        let persisted_epoch = self
```

Comment:
We can match only `(start_offset, aligned_range_start)` and  delay calculating the `persisted_epoch ` when we match to `(None, None)`.